akka-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala

```diff
@@ -10,10 +10,12 @@ import akka.parboiled2._
 import akka.shapeless._
 import akka.http.scaladsl.model._
 
+private[http] case class HeaderParserSettings(uriParsingMode: Uri.ParsingMode = Uri.ParsingMode.Relaxed)
```

Comment:
The default code path is to construct this from `akka.http.server.parsing.uri-parsing-mode` in HttpHeaderParser. But you are right that I can probably get rid of the wrapping by changing the class into a trait.